cpp/src/io/parquet/writer_impl.hpp

```diff
@@ -51,6 +49,25 @@ class parquet_column_view;
 using namespace cudf::io::parquet;
 using namespace cudf::io;
 
+/**
+ * @brief Chunked writer state struct. Contains pieces of information
+ *        needed that span the write() / end() call process.
+ */
+struct pq_chunked_state {
+  /// current write position for rowgroups/chunks
+  std::size_t current_chunk_offset;
+  // special parameter only used by detail::write() to indicate that we are guaranteeing
+  // a single table write.  this enables some internal optimizations.
+  bool single_write_mode;
```

Comment:
Seems like this can be an arg and a non-changing member to impl class.

Reply:
Should I also move current_chunk_offset to impl since it only a single arg left